salt/tests/unit/modules/files/test_metalk8s_kubernetes_utils.yaml

```diff
@@ -278,3 +278,97 @@ get_service_endpoints:
       subsets: null
     raises: True
     result: "Unable to get kubernetes endpoints for my_service in namespace my_namespace"
+
+get_service_ips_and_ports:
+  # 1. Multiples IPs and multiple ports
+  - obj:
+      spec:
+        clusterIPs:
+          - 10.0.0.1
+          - 10.0.0.2
+        clusterIP: 10.0.0.1
+        ports:
+          - name: http-web
+            port: 1234
+          - name: reloader-web
+            port: 1235
+    result:
+      ips:
+        - 10.0.0.1
+        - 10.0.0.2
+      ports:
+        http-web: 1234
+        reloader-web: 1235
+  # 2. Single IP and multiple ports
+  - obj:
+      spec:
+        clusterIPs:
+          - 10.0.0.1
+        clusterIP: 10.0.0.1
+        ports:
+          - name: web
+            port: 2345
+          - name: reloader-web
+            port: 2346
+    result:
+      ips:
+        - 10.0.0.1
+      ports:
+        web: 2345
+        reloader-web: 2346
+  # 3. Only clusterIP and multiple ports
+  - obj:
+      spec:
+        clusterIP: 10.0.0.1
+        ports:
+          - name: web
+            port: 2345
+          - name: reloader-web
+            port: 2346
+    result:
+      ips:
+        - 10.0.0.1
+      ports:
+        web: 2345
+        reloader-web: 2346
+  # 4. No IPs and multiple ports
+  - obj:
+      spec:
+        ports:
+          - name: web
+            port: 1234
+          - name: reloader-web
+            port: 1235
+    result:
+      ports:
+        web: 1234
+        reloader-web: 1235
+  # 5. Unnamed ports
```

Comment:
Can we add a test with mix of unamed and named ports and at least one test with a single port cause here we always have 2